Fix labwc-9999 build failing due to missing libsfdo

BGazotti opened the pull request:
Labwc added support for window icons in https://github.com/labwc/labwc/commit/a745f911697b4877d0edb1bb9aec4f2ff3145d12, using libsfdo.

That feature is enabled by default and will result in a configure error if that library isn't found.

Commits add a very crude but working ebuild for gui-libs/libsfdo and the 'icons' USE flag to labwc to enable optionally building that feature.

Alternatively, we could just add the libsfdo tarball to the ebuild as an exception for the network sandbox and allow labwc's meson scripts to build and bundle it.

BGazotti committed:
gui-libs/libsfdo: new package: gui-libs/libsfdo-0.1.3

BGazotti committed:
gui-wm/labwc: optional window icon support

tfemby commented:
Second on this because of todays commit https://github.com/labwc/labwc/commit/3a4e0cf98115e8eca9f587643d4fea243e5e4741 which version bumped it up to the version you made for the ebuild you made.

Sherman5A commented:
Also labwc's version has been bumped to 0.8.1 https://github.com/labwc/labwc/commit/6bd36e85122101b30a852723b33d72210d45f8a5 So the versioned ebuilds can also be updated.

tfemby commented on gui-libs/libsfdo/libsfdo-0.1.3.ebuild:
```diff
@@ -0,0 +1,24 @@
+# Copyright 2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2-or-later
```
This should be changed to the ISC license as per the licensing policy. Take a look at the main labwc ebuild how it's been done :)

tfemby commented:
Looking through this properly now that I have more time as well, the libsfdo ebuild needs an accompanying metadata.xml.

Testing this locally, I ran `pkgcheck scan` on this the following came up:

```
  MissingRemoteId: missing <remote-id type="freedesktop-gitlab">vyivel/libsfdo</remote-id> (inferred from URI 'https://gitlab.freedesktop.org/vyivel/libsfdo')
  BetterCompressionUri: version 0.1.3: line 8: better compression URI using extension '.tar.bz2' for 'https://gitlab.freedesktop.org/vyivel/libsfdo/-/archive/v${PV}/libsfdo-v${PV}.tar.gz'
  EmptyGlobalAssignment: version 0.1.3: line 17: empty global assignment: DEPEND=""
```

To placate pkgcheck, you could change the .tar.gz to .tar.bz2. The DEPEND and BDEPEND lines aren't necessary so they can simply get removed.

I may also soon send @epsilon-0  an e-mail asking what's happening since this has been a PR for a few months now. I hope they're doing ok.

tfemby commented on gui-libs/libsfdo/libsfdo-0.1.3.ebuild:
```diff
@@ -0,0 +1,24 @@
+KEYWORDS="~amd64"
+
+DEPEND=""
+RDEPEND="${DEPEND}"
```
DEPEND and RDEPEND aren't doing anything so you can get rid of these lines.

tfemby commented on gui-libs/libsfdo/libsfdo-0.1.3.ebuild:
```diff
@@ -0,0 +1,24 @@
+
+DESCRIPTION="A collection of libraries which implement some freedesktop.org specifications"
+HOMEPAGE="https://gitlab.freedesktop.org/vyivel/libsfdo"
+SRC_URI="https://gitlab.freedesktop.org/vyivel/libsfdo/-/archive/v${PV}/libsfdo-v${PV}.tar.gz"
```
pkgcheck is saying that .tar.bz2 is more efficient so it's probably worth grabbing that format.

BGazotti replied:
bz2 is indeed more space-efficient, but decompresses slower. Shouldn't make a difference for such a small file though

tfemby replied:
Agreed, it's probably the least important out of the 3 things.

BGazotti committed:
gui-libs/libsfdo: add metadata.xml

use .bz2 distfile for lower disk and network overhead
update license and copyright information

tfemby reviewed (approved):
Everything looks good to me. Thanks for the changes!

tfemby commented:
Looks like the guru overlay picked up labwc and a couple others wayland compositors.

https://github.com/gentoo/guru/commit/cbbc7f93ecbec73759b5f23dec23fc4671d6bd7a
